include/seqan3/std/type_traits

```diff
@@ -24,6 +24,35 @@ namespace std
  * \brief The \<concepts\> header from C++20's standard library.
  */
 
+// ----------------------------------------------------------------------------
+// remove_cvref_t
+// ----------------------------------------------------------------------------
+#ifndef __cpp_lib_remove_cvref
+/*!\brief Return the input type with `const`, `volatile` and references removed.
+ * \tparam t The type to operate on.
+ */
+template <typename t>
+struct remove_cvref;
+
+//!\cond
+template <typename t>
+    requires true
+struct remove_cvref<t> // needed for gcc-9, it defines std::remove_cvref but does not populate __cpp_lib_remove_cvref
+{
+    using type = std::remove_cv_t<std::remove_reference_t<t>>;
+};
+//!\endcond
+
+/*!\brief Return the input type with `const`, `volatile` and references removed (transformation_trait shortcut).
+ * \tparam t The type to operate on.
+ */
+template <typename t>
+using remove_cvref_t = typename remove_cvref<t>::type;
```

Comment:
```suggestion
template <typename t>
//!\cond
    requires true // needed for gcc-9, it defines std::remove_cvref but does not populate __cpp_lib_remove_cvref
//!\endcond
using remove_cvref_t = typename remove_cvref<t>::type;
```

Apparently this only happens for GCC9 on MacOS

Reply:
This should also fail on gcc9 linux...

Also, in line 21: `#ifndef __cpp_lib_type_identity`
should go down to line 56

Reply:
@marehr That means we need a workaround macro? `__cpp_lib_remove_cvref` not being defined will be fixed in 9.4

Reply:
Not really, this is the patch http://gcc.1065356.n8.nabble.com/committed-0-8-libstdc-Add-update-fix-feature-test-macros-td1682441.html

This is where I found it (remove_cvref has a note that says since 9.4 https://gcc.gnu.org/onlinedocs/libstdc++/manual/status.html)